### ISSUE TITLE
Hi Request  failure not return main_queue 

### ISSUE THREAD

**Issue by orange2003** [MLSDev/TRON#3]
I find int APIRequest.swift 
`dispatch_async(dispatch_get_global_queue(QOS_CLASS_USER_INITIATED, 0), {
                let object : AnyObject
                do {
                    object = try (data ?? NSData()).parseToAnyObject()
                }
                catch let jsonError as NSError {
                    failure?(errorBuilder.buildErrorFromRequest(urlRequest, response: response, data: data, error: jsonError))
                    return
                }

```
            let model: Model.ModelType
            do {
                model = try responseBuilder.buildResponseFromJSON(object)
            }
            catch let parsingError as NSError {
                failure?(errorBuilder.buildErrorFromRequest(urlRequest, response: response, data: data, error: parsingError))
                return
            }
            dispatch_async(dispatch_get_main_queue(), {
                success(model)
            })
        })`
```

`failure?(errorBuilder.buildErrorFromRequest(urlRequest, response: response, data: data, error: parsingError))` not dispatch_get_main_queue why?

**Comment by DenTelezhkin** [comment]
Hey!

Actually right now QOS_CLASS_USER_INITIATED queue happens to be main thread, not background. I'm working to improve calling of success/failure blocks, hopefully with customization available on which queue should completion run. I'll leave this issue open and update it, when feature will be finished.

For now it is safe to assume that failure blocks are being called on main thread.

**Comment by DenTelezhkin** [comment]
Closed by 5639b960e968586d1e24a7adcc6a3420e8648d49.

Thanks for pointing out!
